okhttp/src/main/java/okhttp3/internal/http2/Http2ExchangeCodec.kt

```diff
@@ -106,7 +107,10 @@ class Http2ExchangeCodec(
   }
 
   override fun reportedContentLength(response: Response): Long {
-    return response.headersContentLength()
+    return when {
+      !response.promisesBody() -> 0L
+      else -> response.headersContentLength()
+    }
```

Comment:
I think we don't have to handle chunked (Transfer-Encoding) response unlike HTTP1.

https://tools.ietf.org/html/rfc7540#section-8.1 says `HTTP/2 uses DATA frames to carry message payloads. The "chunked" transfer encoding defined in Section 4.1 of [RFC7230] MUST NOT be used in HTTP/2.`